src/compas_ags/rhino/forceobject.py

```diff
@@ -32,6 +39,122 @@ def inspector_on(self, form):
     def inspector_off(self):
         self.inspector.disable()
 
+    def scale_from_3_points(self):
+        """Scale the ForceDiagram from 3 points
+        """
+        color = Rhino.ApplicationSettings.AppearanceSettings.FeedbackColor
+
+        vertex_xyz = self.artist.vertex_xyz
+        edges = list(self.diagram.edges())
+        anchor_xyz = self.diagram.vertex_attributes(self.artist.anchor_vertex, 'xyz')
+        origin = self.artist.anchor_point
+
+        # select the base point as the anchor point
+        gp = Rhino.Input.Custom.GetPoint()
+        gp.SetCommandPrompt('Select the base point.')
+        gp.Get()
+        if gp.CommandResult() != Rhino.Commands.Result.Success:
+            return False
+        base = gp.Point()
+
+        # get the first reference point
+        gp = Rhino.Input.Custom.GetPoint()
+        gp.SetCommandPrompt('Select the 1st reference point.')
+        gp.DrawLineFromPoint(base, True)
+        gp.Get()
+        if gp.CommandResult() != Rhino.Commands.Result.Success:
+            return False
+        ref1 = gp.Point()
+
+        # get the second reference point
+        gp.SetCommandPrompt('Select the 2nd reference point.')
+
+        def OnDynamicDraw(sender, e):
+            base_ref1 = base.DistanceTo(ref1)
+            base_ref2 = base.DistanceTo(e.CurrentPoint)
+            ratio = base_ref2 / base_ref1
+            for vertex in self.diagram.vertices():
+                xyz = self.diagram.vertex_attributes(vertex, 'xyz')
+                vector = subtract_vectors(xyz, anchor_xyz)
+                vertex_xyz[vertex] = add_vectors(origin, scale_vector(vector, self.artist.scale * ratio))
+            for u, v in iter(edges):
+                e.Display.DrawDottedLine(Point3d(* vertex_xyz[u]), Point3d(* vertex_xyz[v]), color)
+
+        gp.DynamicDraw += OnDynamicDraw
+        gp.Get()
+        if gp.CommandResult() != Rhino.Commands.Result.Success:
+            return False
+        ref2 = gp.Point()
+
+        base_ref1 = base.DistanceTo(ref1)
+        base_ref2 = base.DistanceTo(ref2)
+        ratio = base_ref2 / base_ref1
+        scale_factor = self.artist.scale * ratio
+        self.artist.scale = scale_factor
+
+    def scale_from_4_points(self):
```

Comment:
perhaps this one can indeed be removed

Reply:
i remove scale_from_4_points?

Reply:
I think we can remove this, is not  intuitive plus in rhino the Scale command works exactly as scale from 3 points.

Actually the command "Scale" in rhino asks you: "Choose Base Point or Scale Factor" what solves the problem in only one function.

Reply:
okay so let's remove this

Reply:
yep. I was thinking the same.